operator/v1/types_ingress.go

```diff
@@ -225,6 +241,65 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// Ciphers defines the cipher algorithms used by an IngressController for performing
+// encryption or decryption of network connections. A profile can be specified or a
+// custom strategy of ciphers.
+type Ciphers struct {
+	// Profile is one of three recommended cipher configurations from [1].
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
+	//
+	// +optional
+	Profile CipherProfile `json:"profile,omitempty"`
+	// CustomStrategy consists of a manually defined list of ciphers with each
+	// cipher separated by a ":".
+	//
+	// +optional
+	CustomStrategy *string `json:"customStrategy,omitempty"`
+}
+
+// CipherProfile is a way to specify a supported cipher profile.
+type CipherProfile string
+
+const (
+	// old is a cipher profile that works with all clients back to Windows XP/IE6.
+	CipherProfileOldType CipherProfile = "old"
+	// intermediate is is a cipher profile for services that do not need compatibility
+	// with legacy clients (mostly WinXP), but still need to support a wide range of
+	// clients, this configuration is recommended. It is is compatible with Firefox 1,
+	// Chrome 1, IE 7, Opera 5 and Safari 1.
+	CipherProfileIntermediateType CipherProfile = "intermediate"
+	// modern is a cipher profile for services that do not need backward compatibility,
+	// the parameters below provide a higher level of security. This profile is
+	// compatible with Firefox 27, Chrome 30, IE 11 on Windows 7, Edge, Opera 17,
+	// Safari 9, Android 5.0, and Java 8.
+	CipherProfileModernType CipherProfile = "modern"
+)
+
+// SecurityProtocol defines one or more security protocols used by
+// an IngressController to secure network connections.
+type SecurityProtocol struct {
+	MinimumVersion SecurityProtocolVersion
+	MaximumVersion SecurityProtocolVersion
```

Comment:
These fields are missing json tags.  Is a range appropriate?  Can we be sure that no one will need to enable `TLSv1.1` and `TLSv1.3` but disable `TLSv1.3`, for example?  Is it generally accepted and understood that `SSLv3` is less than `TLSv1.0`?

Reply:
@Miciah thanks for reviewing the PR. `MinimumVersion` and `MaximumVersion` map to HAProxy config file [directives](https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#5.1-ssl-max-ver) `ssl-min-ver` and `ssl-max-ver`. According to HAProxy docs, these are the 2 recommended directives for tuning ssl versions. I believe the common use case is to satisfy cybersecurity baseline requirements by A) requiring a specific protocol version (setting `MinimumVersion` and `MaximumVersion`) or B) requiring a minimum protocol version by specifying `MinimumVersion`.

> Is it generally accepted and understood that SSLv3 is less than TLSv1.0?

This is a good point. I think we should address this in documentation. Let me know if you have any suggestions on how to address this problem.

Reply:
> `MinimumVersion` and `MaximumVersion` map to HAProxy config file [directives](https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#5.1-ssl-max-ver) `ssl-min-ver` and `ssl-max-ver`.

HAProxy 1.8 and and Envoy 1.10 allow a range to be specified, but Nginx and AWS ELBs allow versions to be selectively enabled or disabled—and we don't know what future versions of HAProxy or Envoy may allow.  This presents a trade-off: Do we limit configurability for admittedly hypothetical implementations, or do we make the API more general and then validate the given parameters against the actual implementation?

Reply:
IMO the common use case is to specify a minimum protocol version, which maps nicely to the proposed `MinimumVersion` and `MaximumVersion ` fields.